### ISSUE TITLE
Feature Request: Check pkgrepo state deprecation

### ISSUE THREAD

**Issue by jbouter** [warpnet/salt-lint#38]
### Preface
Kindly use the [check for state.cmd deprecation](https://github.com/roaldnefs/salt-lint/pull/37) as an example of adding a check. #37 has to be merged first.

### Description

Add the possibility to check for deprecation within the [pkgrepo state](https://docs.saltstack.com/en/latest/ref/states/all/salt.states.pkgrepo.html)

The following options have been deprecated:
* `refresh_db`

The following options should be used:
* `refresh`

Example(s) of an **improperly** configured state:
```code
foo:
  pkgrepo.managed:
    - humanname: Personal repo for foo
    - baseurl: https://mydomain.tld/repo/foo/$releasever/$basearch
    - gpgkey: file:///etc/pki/rpm-gpg/foo-signing-key
    - gpgcheck: 1
    - refresh_db: True
```

Example(s) of **properly** configured state:
```code
foo:
  pkgrepo.managed:
    - humanname: Personal repo for foo
    - baseurl: https://mydomain.tld/repo/foo/$releasever/$basearch
    - gpgkey: file:///etc/pki/rpm-gpg/foo-signing-key
    - gpgcheck: 1
    - refresh: True
```
It has been deprecated since _version 2018.3.0_

### Fix requirements
A fix for this issue should (at least):
* Notify the user of using a deprecated option
* Notify the user since what version the option will be or has been deprecated
* A suggestion on what option to replace it with
* Contain unit tests for the check

**Comment by myii** [comment]
@jbouter @roaldnefs I really like the idea of all of these deprecation checks that have been proposed.  I'd just like to share an angle from a SaltStack Formulas perspective.  We've had detailed discussions about how to handle deprecations and Salt versions that we have to support.  Our eventual conclusion was that we would keep in sync with upstream Salt's support cycle.  At first, we started from this page:

* https://www.saltstack.com/product-support-lifecycle/

In discussions with the Salt team, we were able to refine that into a simpler rubric: the latest _three_ versions of Salt are supported at any given time.  So until the next major release, that's:

* `2017.7`
* `2018.3`
* `2019.2`

You'll see that's what we've been using for all of our Kitchen/InSpec-based testing via. Travis (in the formulas that have been converted).  We also test against `develop`, to be ready for the next release in advance.

So how does this all apply here to this discussion?  We've found that we have to support deprecated options while a given version of Salt is still supported.  I believe `salt-lint` should consider doing the same.  _I'm not saying that these rules shouldn't be implemented_; they'll be an excellent addition to this tool.  Rather, my suggestion is to allow setting the range of Salt versions to be supported in the `.salt-lint` file.  Then automatically test rules based on whether the rule is "deprecated" based on the user's circumstances or not.  So that would mean tracking the _minimum_ and _maximum_ Salt versions for each rule and then comparing that when it's time to test for it.  Essentially, this can be done automatically, rather than placing the burden of enabling/disabling rules on the end-user, depending on what versions of Salt they have to support, in a given infrastructure setting.

**Comment by jbouter** [comment]
Thanks for your feedback @myii! We should indeed not throw errors (or when used inside a CI-pipeline, fail) for functionality that'll be deprecated in the future (from the standpoint of the oldest supported release). I would suggest, however, to still check for these to-be-deprecated features and throw a warning (not affecting the exit code of `salt-lint`) much like the warnings within the SaltStack documentation is shown.

It would be nice indeed to have a minimal and maximum version configurable to be checked against in the `.salt-lint` configuration file. Regardless of SaltStack support, linux distributions may have their own support cycle that we should take into account (for example, Ubuntu 18.04 will have support till 2023, and contains version `2017.7`.

**Comment by roaldnefs** [comment]
Closed in favour of #204
